packages/blade/src/components/Checkbox/CheckboxInput.web.tsx

```diff
@@ -0,0 +1,68 @@
+/* eslint-disable @typescript-eslint/ban-types */
+import type { CSSObject } from 'styled-components';
+import styled from 'styled-components';
+import type { Theme } from '../BladeProvider';
+import { screenReaderStyles } from '../VisuallyHidden';
+import type { CheckboxIconProps } from './CheckboxIcon';
+import type { InputProps } from './useCheckbox';
+import { getIn, makeMotionTime } from '~utils';
+
+type HoverProps = Pick<CheckboxIconProps, 'isNegative' | 'isDisabled'> & {
+  isChecked: boolean;
+};
+
+const variants = {
+  default: {
+    background: {
+      checked: 'colors.brand.primary.600',
+      unchecked: 'colors.brand.gray.400',
+    },
+  },
+};
+
+const getHoverStyles = ({
+  theme,
+  isDisabled,
+  isNegative,
+  isChecked,
+}: { theme: Theme } & HoverProps): CSSObject => {
+  if (isDisabled || isNegative) return {};
+
+  const checked = isChecked ? 'checked' : 'unchecked';
+  const backgroundColor = variants.default.background[checked];
+
+  return {
+    backgroundColor: getIn(theme, backgroundColor),
+    transitionTimingFunction: theme.motion.easing.standard.effective as string,
+    transitionDuration: makeMotionTime(theme.motion.duration['2xquick']),
+  };
+};
+
+const StyledInput = styled.input<HoverProps>(({ theme, isChecked, isDisabled, isNegative }) => ({
+  ...screenReaderStyles,
+  '&:focus + div': {
```

Comment:
isn't this risky? what if we replace the underlying wrapper component and it doesn't renders `div`? will this still work?

Reply:
Not risky. I will say fragile. Our tests can catch it tho. 

The workaround is more code tho. Like using useFocusWithin hook to detect when the input is focused then track the focused state and depending on that draw the focus indicator.